syntax = "proto3";

package chat;

// Mensagem enviada de um cliente para o servidor
message ChatMessage {
  string username = 1;
  string message = 2;
}

// Requisição de entrada de chat
message JoinRequest {
  string username = 1;
}

// Resposta quando o usuário entra no chat
message JoinResponse {
  string welcome_message = 1;
}

// Serviço de chat com dois métodos
service ChatService {
  // Cliente entra no chat
  rpc JoinChat(JoinRequest) returns (JoinResponse);

  // Cliente envia uma mensagem para o chat (stream bidirecional)
  rpc Chat(stream ChatMessage) returns (stream ChatMessage);
}


